src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1221,6 +1221,44 @@ void MPS::full_state_vector_internal(cvector_t& statevector,
   statevector = reverse_all_bits(temp_statevector, num_qubits);
 }
 
+void MPS::get_amplitude_vector(const reg_t &base_values, cvector_t &amplitude_vector) {
+  move_all_qubits_to_sorted_ordering();
```

Comment:
Insertion of swaps should be avoided when possible, and can be avoided in this case

Reply:
How do you suggest to avoid the swaps here?

Reply:
If the internal order is: qubit 1 in position 0, qubit 2 in position 1, qubit 0 in position 2. And the user asks for the amplitude of base state 011 (1 for qubit 0, 1 for qubit 1, 0 for qubit 2). Then transform the base state to another base state, according to the internal ordering: 101. Now contract without swaps.

Reply:
Done. Thanks for this improvement!